datacube/index/abstract.py

```diff
@@ -912,16 +915,29 @@ def archive_less_mature(self, ds: Dataset, delta: int = 500) -> None:
         for lm_ds in less_mature_ids:
             _LOG.info(f"Archived less mature dataset: {lm_ds}")
 
-    def find_less_mature(self, ds: Dataset, delta: int = 500) -> Iterable[Dataset]:
+    def find_less_mature(self, ds: Dataset, delta: Union[int, bool] = 500) -> Iterable[Dataset]:
         """
         Find less mature versions of a dataset
 
         :param Dataset ds: Dataset to search
-        :param int delta: millisecond delta for time range
+        :param Union[int,bool] delta: millisecond delta for time range.
+        If True, default to 500ms. If None or False, do not find or archive less mature datasets.
+        Bool value accepted only for improving backwards compatibility, int preferred.
         :return: Iterable of less mature datasets
         """
-        less_mature = []
-        assert delta >= 0
+        if isinstance(delta, int):
```

Comment:
I got recently bit by this, but in Python `True` and `False` are both "int" as well as "bool", so `elif` won't ever trigger.

```python
[(isinstance(b, int), isinstance(b, bool), b is True, b is False)
 for b in [0,1, False, True]]
```

Reply:
yes, I have just come to the same realisation!